### PR TITLE
Feature: GitHub create fork command

### PR DESCRIPTION
**Pull request by asfaltboy**
-----

The maintainers of this repo require that all pull request submitters agree and adhere to the following:

- [x] I have read and agree to the [contribution guidelines](https://github.com/divmain/GitSavvy/blob/master/CONTRIBUTING.md).
 _(required)_
- [x] All related documentation has been updated to reflect the changes made. _(required)_
- [x] My commit messages are cleaned up and ready to merge. _(required)_

The maintainers of this repository require you to select the semantic version type that
the changes in this pull request represent.  Please select one of the following:
- [ ]  major
- [ ]  minor
- [ ]  patch
- [ ]  documentation only

### CODE REVIEW

**Comment by randy3k** [comment]
It works, but we might need to throw a better message when a user didn't specify the GitHub token in the settings.

**Comment by randy3k** [comment]
Btw, I want to have something like this for ages, just don't have the momentum to look at the github api on doing it.

**Comment by asfaltboy** [comment]
Thanks for the feedback @randy3k ! I just really wanted to fix some pesky bugs in [divmain/maintainerd][1] and was surprised we don't have this. I'll add a short message and link to [our "GitHub setup" docs][2].

[1]: https://github.com/divmain/maintainerd
[2]: https://github.com/divmain/GitSavvy/blob/master/docs/github.md#setup

File: github/github.py [hunk 1]
```diff
@@ -13,6 +13,14 @@
 from ..core.exceptions import FailedGithubRequest
 
 GITHUB_PER_PAGE_MAX = 100
+GITHUB_ERROR_TEMPLATE = "Error {action} Github: {payload}"
+AUTH_ERROR_TEMPLATE = """Error {action} Github, access was denied!
+
+Please ensure you have crated a Github API token and added it to
```
Comment (randy3k) [inline github/github.py]: created?

File: github/github.py [hunk 2]
```diff
@@ -121,24 +129,48 @@ def get_api_fqdn(github_repo):
...
+        query_params=interwebs.urlencode(kwargs))
+
+
+def validate_response(response, method="GET"):
+    action = {"GET": 'querying', "POST": 'posting to'}[method]
+
+    if response.status in [401, 403]:
+        raise FailedGithubRequest(AUTH_ERROR_TEMPLATE.format(action=action))
+
+    if 200 > response.status > 299 or not response.is_json:
```
Comment (randy3k) [inline github/github.py]: Should it be `response.status < 200 or response.status > 299`?

**Comment by asfaltboy** [comment]
Nice catch, thanks! Kindly review all's wel

File: github/commands/create_fork.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    def run(self):
+        sublime.set_timeout_async(self.run_async, 0)
+
+    def run_async(self):
+        base_remote = github.parse_remote(self.get_integrated_remote_url())
+        sublime.status_message(START_CREATE_MESSAGE.format(repo=base_remote))
+        result = github.create_fork(base_remote)
+        sublime.status_message(END_CREATE_MESSAGE)
+        util.debug.add_to_log(("github: fork result:\n{}".format(result)))
+        self.window.show_quick_panel(
```
Comment (randy3k) [inline github/commands/create_fork.py]: I quite like this kind of prompt, way better than `sublime.ok_cancel_dialog`. I think we should create our version of `ok_cancel_prompt` based on `quick_panel`. But it is a story of another time.

**Pull request merged**